SimplenoteTests/NSStringSimplenoteTests.swift

```diff
@@ -6,6 +6,17 @@ import XCTest
 //
 class NSStringSimplenoteTests: XCTestCase {
 
+    /// Verifies that `byEncodingNonAlphanumerics` effectively escapes all of the non alphanumeric characters
+    ///
+    func testByEncodingNonAlphanumericsPercentEncodesAllOfTheNonAlphanumericCharactersInTheReceiver() {
+        let sample = "1234567890!@#$%^&*()-_+[]';./,qwertyuiopasdfghjkl;'zxcvbnm,./🔥😂😃🤪👍🦆🏴‍☠️☝️😯"
+        let escaped = sample.byEncodingNonAlphanumerics ?? ""
+        let escapedSet = CharacterSet(charactersIn: escaped)
+        let expectedSet = CharacterSet(charactersIn: "%").union(.alphanumerics)
+
+        XCTAssertTrue(expectedSet.isSuperset(of: escapedSet))
+    }
```

Comment:
noting that this test is mostly asserting a behavior of a built-in function which presumably already has rock-solid and comprehensive tests

Reply:
Oh yeah, this was mostly for myself (to ensure... the API does what I intended to actually do!), rather than manually verifying 😄